Library/Homebrew/dependency.rb

```diff
@@ -28,8 +30,26 @@ def hash
     name.hash ^ tags.hash
   end
 
+  def use_best_spec!
```

Comment:
Newest is not what we actually mean here, see the comment to `Formula#best_upgrade_spec`